### PR TITLE
Add themes/base.theme.bash to clean files

### PR DESCRIPTION
**Pull request by marcospereira**
## Description

Review with [?w=1](https://github.com/Bash-it/bash-it/pull/1785/files?w=1).

- Add themes/base.theme.bash to clean files
- Run code fomatter for themes/base.theme.bash
- Fix shellcheck warnings for themes/base.theme.bash
- Fix shellcheck header script to consider multiple lines

## Motivation and Context

I've picked this file because it is the file with most changes (commits, at least) based on:

    git log --name-only --pretty="format:" | grep -v -e "^[[:space:]]*$" | sort | uniq -c | sort

I thought about adding other files, but that would probably make it harder to review.

## How Has This Been Tested?

Besides running the bats tests locally, I've manually tested that the most significant in isolation (checking they are resulting in the same effect as before).

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] File linting

## Checklist:

- [x] My code follows the code style of this project.
- [x] If my change requires a change to the documentation, I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] If I have added a new file, I also added it to ``clean_files.txt`` and formatted it using ``lint_clean_files.sh``.
- [x] I have added tests to cover my changes, and all the new and existing tests pass.

### CODE REVIEW

File: hooks/dot-bash.sh [hunk 1]
```diff
@@ -11,8 +11,8 @@ for file in "$@"; do
 
 	# Confirm expected schellcheck header
 	#
-	LINE1="$(head -n 1 "${file}")"
-	if [[ "${LINE1}" != "# shellcheck shell=bash" ]]; then
+	HEADER="$(head -n 2 "${file}")"
```
Comment (marcospereira) [inline hooks/dot-bash.sh:14]: I'm using two lines here, which should be okay.

But maybe we don't even want to check all bash files listed in `clean_files.txt`? It seems to me this was intended to be used only for included files.

Comment (NoahGorny) [reply hooks/dot-bash.sh:14]: .bash files are always included in Bash-it, and not invoked directly. `theme.base.bash` is no exception. You should remove the change in the script and remove the shebang from the file as well

**Review by NoahGorny — COMMENTED** [review]
Wow @marcospereira, amazing job :smile:
I will take a look in the weekend, but I want to say thank you for taking your time to put an effort to improve bash it :heart:

File: themes/base.theme.bash [hunk 2]
```diff
@@ -1,4 +1,5 @@
 #!/usr/bin/env bash
+# shellcheck shell=bash
```
Comment (davidpfarrell) [inline themes/base.theme.bash:2]: Just chiming in to confirm that we need to remove the 1st line here, leaving only the shellcheck statement.
(and of course also reverting the changes to `dot-bash.sh`

**Comment by marcospereira** [comment]
Thanks for reviewing, @NoahGorny and @davidpfarrell. I've removed the changes in `dots-bash.sh`.

**Review by NoahGorny — APPROVED** [review]
Amazing work @marcospereira!
Thank you for your effort!

**Comment by marcospereira** [comment]
Thanks, @NoahGorny. So, I can invest some more time in this cleanup task, but perhaps just prioritizing files that have many changes isn't the better way to do it, right?

Are there any specific areas where you think it would be more relevant now?

**Comment by NoahGorny** [comment]
> Thanks, @NoahGorny. So, I can invest some more time in this cleanup task, but perhaps just prioritizing files that have many changes isn't the better way to do it, right?
> 
> Are there any specific areas where you think it would be more relevant now?

I think that going after the core files, like the file you fixed here, is a great idea. Just make sure there is no pending PRs for the file, and lint away :smile:
For hints, I think `bash_it.sh` is a challenging file worth cleaning up, also the `lib` directory contains important files.
Thank you for doing this @marcospereira !

**Pull request merged**